streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -577,6 +583,7 @@ public void suspend() {
     // visible for testing
     void suspend(final boolean clean,
                  final boolean isZombie) {
+        final Map<TopicPartition, Long> partitionTimes = extractPartitionTimes();
```

Comment:
I remember now -- can we add a comment to explain that we need to get `partitionTimes` before we `closeTopology()` (sorry for my previous comment -- forgot about that)

Reply:
Cool, got it done.